system/jhtree/keybuild.cpp

```diff
@@ -605,7 +605,20 @@ class CKeyBuilder : public CInterfaceOf<IKeyBuilder>
     virtual unsigned __int64 getOffsetBranches() const override { return offsetBranches; }
     virtual unsigned __int64 getBranchMemorySize() const override { return indexCompressor->queryBranchMemorySize(); }
     virtual unsigned __int64 getLeafMemorySize() const override { return indexCompressor->queryLeafMemorySize(); }
-
+    virtual unsigned __int64 getStatistic(StatisticKind kind) const override
+    {
+        switch (kind)
+        {
+        case StNumLeafCacheAdds:
+            return numLeaves;
+        case StNumNodeCacheAdds:
+            return numBranches;
+        case StNumBlobCacheAdds:
```

Comment:
future: add other stats (offsetBranches, branchMemorySize, leadMemorySize) to stats mapping and collect these via gather stats too.

Reply:
Future: https://hpccsystems.atlassian.net/browse/HPCC-33279